core/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -145,10 +142,15 @@ object MediaRange {
       override def render(writer: Writer, range: MediaRange): writer.type =
         range.render(writer)
 
-      override def eqv(x: MediaRange, y: MediaRange): Boolean =
-        x.equals(y)
+      override def compare(x: MediaRange, y: MediaRange): Int = {
+        def orderedSubtype(a: MediaRange) = a match {
+          case mt: MediaType => mt.subType
+          case _ => ""
+        }
```

Comment:
The awkwardness of this flows from the awkwardness of the types.  We should do better here.  (Sealed concrete types? :confused:)

@cquiroz mentioned he's been thinking about this.